### ISSUE TITLE
[Bug]: Recently Added shouldn't rely on Update but Create timestamp

### ISSUE THREAD

**Issue by anarcat** [navidrome/navidrome#2650]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.50.1

### Current Behavior

Modifications to (say) an album cover brings said album back into the top of the "Recently Added" queue, even though it might be an old album.

### Expected Behavior

Instead, "Recently Added" should only rely on the creation (or import?) date of the file that's being processed.

### Steps To Reproduce

 1. import two albums in Navidrome, with file timestamps far enough apart to make them sortable
 2. modify the old one by, say, adding an album cover

Expected result: the newer album before step 2 should still show up as the newest.

Actual result: modified album bumps the newer album.

### Environment

```markdown
- OS: Debian GNU/Linux bookworm, Docker container
- Browser: Firefox ESR, but irrelevant
- Client: Supersonic or web
```

### How Navidrome is installed?

Docker

### Configuration

```toml
---
# managed by Puppet, local changes will be lost
version: "3"

volumes:
  navidrome-data:

services:
  navidrome:
    image: deluan/navidrome:latest
    user: 10000:10000  # should be owner of volumes
    ports:
      - "127.0.0.1:4533:4533"
    restart: unless-stopped
    environment:
      ND_SCANSCHEDULE: 1h
      ND_LOGLEVEL: info
      ND_SESSIONTIMEOUT: 24h
      ND_BASEURL: ""
      ND_RECENTLYADDEDBYMODTIME: "true"
      ND_LASTFM_ENABLED: "false"
      ND_AUTOIMPORTPLAYLISTS: "false"
      ND_ENABLESHARING: "true"
    volumes:
      - "navidrome-data:/data"
      - "/srv/mp3:/music/mp3:ro"
```
```


### Relevant log output

```shell
I perused the logs and didn't find much interesting, so I do not believe that is relevant.
```

### Anything else?

At first I thought this was because Navidrome only held a single timestamp for albums, but after inspecting the sqlite database (after fear of losing my ratings and play counts, which I *think* i did?), I noticed there was a `created_at` AND `updated_at` column, so that data is actually there, just not used correctly, I believe.

But here's two albums from the database that are sorted, IMHO, backwards:

```
sqlite> select name,artist,created_at,updated_at from album where name like '%American III: Solitary%';
American III: Solitary Man|Johnny Cash|2021-07-04T00:51:17.930079536Z|2023-03-09T22:58:57.431454258Z
sqlite> select name,artist,created_at,updated_at from album where name like '%Vertiges%';
Vertiges|zouz|2022-06-01T20:11:36.78922778Z|2022-06-01T19:44:39.486551366Z
```

There, "Johnny Cash" gets sorted first, as more "Recently Added" than "zouz".

Now of course, one fix to this is to run an SQL query updating my entire database to wipe off the differnet `update_at` timestamps, or at least tweak the ones that are incorrect, but that seems like shoving a problem under the rug...

Besides, maybe we *could* have a Recently Updated view as well? That would be interesting, listing the albums that have a different `created_at` and `updated_at` timestamp! :) Unfortunately, it looks like all those timestamps differ here, sometimes by only a few seconds...

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by anarcat** [comment]
In my case, the workaround was relatively simple: I added a *lot* (350+) of covers all at once, in a short timeframe, so I could rewrite the `update_at` fields on those relatively easily with this query:

```
UPDATE album SET updated_at = created_at WHERE updated_at > '2023-03-06' AND updated_at < '2023-03-10' AND created_at < '2023-03-01';
```

But of course, if I fix an old cover again, that poor old album is going to come back up the list and try to play with the young kids and everyone is going to look at him sideways and think "ok boomer" and release a barely hidden sigh that will just make everyone really uncomfortable...

**Comment by metalheim** [comment]
You have set ND_RECENTLYADDEDBYMODTIME to true which is not what you want from what your description reads.

https://www.navidrome.org/docs/usage/configuration-options/#advanced-configuration

The functionality you describe is implemented with #2553 as long as you dont Set ND_RECENTLYADDEDBYMODTIME (default is False)

**Comment by anarcat** [comment]
Hmm... well that setting doesn't do what I thought it did though: I thought that would set the `created_at` field on import, but from what you seem to be saying here, it influences how the albums are *displayed*?

the `created_at` fields are correctly set here, just to be really clear.

**Comment by deluan** [comment]
> but from what you seem to be saying here, it influences how the albums are _displayed_?

That's correct. It is used for controlling how to display the Recent Added Albums view.

**Comment by anarcat** [comment]
> That's correct. It is used for controlling how to display the Recent Added Albums view.

okay, so couldn't this be made in reverse, like at import time, if that setting is on, use the file mtime as the `created_at` field instead of `updated_at`? for me this thing is completely setup backwards, but maybe that's just me. :)

**Comment by deluan** [comment]
I think the fields should be used for what they mean: `created_at` for when the file was created/first imported (see #1350), and `updated_at` for when was the last time the record was changed in the DB. Inverting these two can cause confusion when maintaining the codebase, and it is just a matter of displaying sorted by one or the other, with no need to change their semantics.

Also, making this distinction at scan time, means that if you want to change it, Navidrome needs to rescan everything again - with a full scan - which can take days sometimes for some large libraries.

**Comment by anarcat** [comment]
i see. so what are my options here?

i guess i could remap the `created_at` timestamp to `updated_at` if the latter is lower? would that work?

**Comment by deluan** [comment]
> Instead, "Recently Added" should only rely on the creation (or import?) date of the file that's being processed.

That should be the behaviour if you remove `ND_RECENTLYADDEDBYMODTIME: "true"`. The default behaviour (`false`) is to use import date (or birth date as per #1350)

**Comment by anarcat** [comment]
> That should be the behaviour if you remove `ND_RECENTLYADDEDBYMODTIME: "true"`. The default behaviour (`false`) is to use import date (or birth date as per #1350)

okay well maybe that's where my confusion comes from: I thought `ND_RECENTLYADDEDBYMODTIME` was a feature flag that enabled the "birth date used on import" defined in #1350 - those two are actually unrelated?

**Comment by deluan** [comment]
> ND_RECENTLYADDEDBYMODTIME was a feature flag that enabled the "birth date used on import" defined in #1350

No it is not!

`ND_RECENTLYADDEDBYMODTIME` selects whether to use `created_at` (false, default) or `updated_at` (true) for sorting the Recently Added albums view. This was implemented by #822 a couple of years ago.

#1350 (fixed by #2553) was released in 0.50.0, which replaces import date with birth date, on systems that support it, as that's a sensible thing to do (use the actual file creation date instead of the import date)

**Comment by anarcat** [comment]
Ookay... I completely misread this! thanks for the clarification...

is there a way to *rescan* files so that their creation date is retrofitted back into the `created_at` field?

**Comment by deluan** [comment]
Try a full rescan. If that doesn't work, there's no other option except for re-adding all your files, which currently will make you lose your playlists and favourites/ratings/play counts :/

**Comment by certuna** [comment]
i noticed to that if the song already has a `created_at` timestamp, a full rescan will not refresh it with the file creation date.

**Comment by deluan** [comment]
This is being handled in #2709

**Comment by deluan** [comment]
Closing as this is not an issue, and making the full rescan reimport all timestamps is being handled in #2709

**Comment by Szeraax** [comment]
I'm actually going to reopen this and we can close it once #2709 is merged and officially released. That way the author can get notified of the fix being available.

**Comment by mattneub** [comment]
@deluan If I create a track outside of the knowledge of Navidrome, and then, say, a year later I finally get around to adding it to Navidrome's "field of view", the album of that track does _not_ appear in the Recently Added Albums list. This is because Navidrome is looking at the date of creation of the track, not the date at which the album suddenly appeared in the music library. This is seems wrong. An album should be regarded as recently created if the album was recently added to the database; it should have nothing to do with how old or new the _tracks_ are. — I make this comment here because it seems to be part of the same discussion, but I'll be happy to open it as a separate issue if you prefer. Thx.

**Comment by deluan** [comment]
@mattneub What you want is the default behaviour, with `RecentlyAddedByModTime=false`. Did you change this option?

**Comment by mattneub** [comment]
I didn't! I'll do that right now. Thank you!

**Comment by oransterf** [comment]
> @mattneub What you want is the default behaviour, with `RecentlyAddedByModTime=false`. Did you change this option?

@deluan - I am looking for the exact opposite behavior. I got confused with all the configuration options so I will ask you instead. What are the correct settings for navidrome to consider recently added as when the file itself (not import) was created (and not updated, only created)? 

Whenever I change tags on the music files themselves they appear as recently added with my current configuration.

Thank you

**Comment by deluan** [comment]
> What are the correct settings for navidrome to consider recently added as when the file itself (not import) was created (and not updated, only created)?

This is only possible if your FS supports birth time. If you are using Linux, use `stat` to check that. You should see a line saying `Birth`

Example of FS that does support it:
```
$ stat docker-compose.yml
  File: docker-compose.yml
  Size: 484       	Blocks: 8          IO Block: 4096   regular file
Device: 254,1	Inode: 750         Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/corintio)   Gid: ( 1000/corintio)
Access: 2024-12-21 15:06:32.012000000 -0800
Modify: 2024-11-24 11:03:32.700000000 -0800
Change: 2024-11-24 11:03:32.820000000 -0800
 Birth: 2024-11-24 10:34:49.188000000 -0800
```

If it is supported, set `RecentlyAddedByModTime=false`. If not, there's nothing I can do about it, and the option will fallback to using the time the file was added to the DB.
